### ISSUE TITLE
Implement proper training setup for transformers🤖

### ISSUE THREAD

**Issue by KarelZe** [KarelZe/thesis#253]
- [x] Clip outliers in dataset
- [x] Add gradient norm
- [x] Unscale gradient
- [x] Add learning rate scheduler with learning rate warmup
- [x] Removed exp. weighting
- [x] Fixes a bug when saving compiled model 
- [x] Adds support for pytorch 2.0
- [x] Allign search space with https://arxiv.org/pdf/2106.11959.pdf 

Experiemnts tracked here: https://wandb.ai/fbv/thesis/reports/FT-Transformer-ISE-downsampled--Vmlldzo0MDQzOTgy
